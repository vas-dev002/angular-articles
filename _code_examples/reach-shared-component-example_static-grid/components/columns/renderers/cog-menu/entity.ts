import { ICellRendererParams } from "@ag-grid-community/core";

export interface ICogMenuEvent {
	params: ICellRendererParams;
	rowIndex: number;
}
export interface ICogMenuItem {
	css?: string;
	hide?: (event: ICogMenuEvent) => boolean;
	name: string;
	onClick: (event: ICogMenuEvent) => void;
}
export interface ICellCogMenuParams extends ICellRendererParams {
	menuItems: ICogMenuItem[];
	isDisabled?: (cellData: unknown) => boolean;
	getCogMenuTooltip?: (cellData: unknown) => string;
}
