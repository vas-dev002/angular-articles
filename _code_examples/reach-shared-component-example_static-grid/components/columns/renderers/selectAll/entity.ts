import { IHeaderParams } from "@ag-grid-community/core";
import { EventEmitter } from "@angular/core";

export interface ISelectAllMethods {
	selectAll: () => void;
	deselectAll: () => void;
	getRowsCount: (excludeDisabledRows: boolean) => number;
	getSelectedCount: () => number;
	onSelectionChanged: EventEmitter<Object[]>;
}

export interface ISelectAllParams extends IHeaderParams {
	suppressHeaderCheckBox?: boolean;
	selectMethods: ISelectAllMethods;
}
