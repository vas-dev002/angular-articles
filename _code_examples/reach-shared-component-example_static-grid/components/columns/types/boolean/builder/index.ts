export * from "./boolean-filtering-options";
export * from "./boolean-grouping-options";
export * from "./boolean-sorting-options";
export * from "./boolean-tooltip-options";
export * from "./boolean-value-formatter-options";
export * from "./boolean-value-getter-options";
