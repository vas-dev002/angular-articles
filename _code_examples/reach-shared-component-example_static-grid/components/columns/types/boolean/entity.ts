export interface BooleanGroupParams {
	value: string;
}
