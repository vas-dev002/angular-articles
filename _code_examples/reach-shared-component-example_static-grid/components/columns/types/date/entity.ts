import { ValueFormatterParams } from "@ag-grid-community/core";

export interface IDateValue {
	date: string;
}

export interface IDateGroupKeyParams {
	value: string;
	date: string;
}

export interface IDateValueFormatterParams extends ValueFormatterParams {
	value: string | IDateValue;
}
