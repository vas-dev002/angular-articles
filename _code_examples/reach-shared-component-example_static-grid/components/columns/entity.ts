import { Type, Injector } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { ColDef } from "@ag-grid-community/core";
import { IColumnMetaSG } from "../main/entity";
import { StaticGridComponent } from "../main/comp";
import { ColumnDisplayType } from "./types";
import { IColDefAbstractFactory } from "./types/base/base-abstract-factory/models";
import { ObjectDetailsType } from "@evergreen-shared/entity";

export interface ITypeControllerSG {
	renderComponent: Type<ICellRendererAngularComp>;
	columnDefinition(): ColDef;
}

export interface IFilterOptionsSG {
	text: string;
	value: unknown;
	props?: Object;
	sortOrder?: number;
}

export interface ITypeControllerParams {
	meta: IColumnMetaSG;
	grid: StaticGridComponent;
	customOptions: Object;
	injector?: Injector;
}

export interface ILinkMeta {
	uRLColumnName: string;
	objectType: string;
	objectTypeColumnName: string;
	labelColumnName: string;
	keyColumnName: string;
	parameters: string;
	urlParameters: string[];
	renderLinkProperty: string;
}

export interface IOutcomeComparatorValue {
	status: number;
	text: string;
	tooltip: string;
}

export enum ApplicationForwardPathTypeEnum {
	None = 0,
	Retire = 1,
	Keep = 2,
	ForwardPath = 3,
	Uncategorized = 4
}

export enum ApplicationAssociationTypeEnum {
	None = 0,
	Dynamic = 1,
	StickyInclude = 2,
	StickyExclude = 3,
	ForwardPathed = 4,
	Mixed = 5
}

export enum StringValueGetterMode {
	String,
	SelectedItem,
	Params
}

export interface ITypedController {
	key: ColumnDisplayType;
	factory: Type<IColDefAbstractFactory>;
}

export enum TaskFlags {
	IsBucketTask = "IsBucketTask",
	IsReadOnlyTask = "IsReadOnlyTask",
	IsSystemTask = "IsSystemTask"
}

export interface IDateTimeEditorChanges {
	taskId: number;
	projectId: number;
	id: number;
	objectType: ObjectDetailsType;
	date: string;
	postBackParameters: { [key: string]: unknown };
}

export interface ICapacityDateTimeEditorChanges extends IDateTimeEditorChanges {
	slotId: number;
	slotName: string;
}
