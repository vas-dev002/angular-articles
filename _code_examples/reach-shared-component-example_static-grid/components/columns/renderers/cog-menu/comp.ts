import { Component } from "@angular/core";
import { ICellRendererAngularComp } from "@ag-grid-community/angular";
import { RowNode } from "@ag-grid-community/core";
import { ICellCogMenuParams, ICogMenuItem } from "./entity";

@Component({
	selector: "child-cell",
	templateUrl: "tpl.html"
})
export class CogMenuSGComponent implements ICellRendererAngularComp {
	public permissionStatus: boolean;
	public params: ICellCogMenuParams;
	public items: ICogMenuItem[];
	public isOpened = false;
	private _getIndex(): number {
		let index = this.params.rowIndex;
		this.params.api.forEachNodeAfterFilterAndSort((rowNode: RowNode, rowIndex: number) => {
			if (this.params.node.id === rowNode.id) {
				index = rowIndex;
			}
		});
		return index;
	}

	public agInit(params: ICellCogMenuParams): void {
		this.params = params;
		this.items = params.menuItems;
	}

	public refresh(): boolean {
		return false;
	}

	public show(item: ICogMenuItem) {
		return item.hide && item.hide({ params: this.params, rowIndex: this._getIndex() });
	}

	public onClick(item: ICogMenuItem) {
		item.onClick({ params: this.params, rowIndex: this._getIndex() });
	}

	public get isDisabled(): boolean {
		return !!this.params && !!this.params.isDisabled && this.params.isDisabled(this.params.data);
	}
}
