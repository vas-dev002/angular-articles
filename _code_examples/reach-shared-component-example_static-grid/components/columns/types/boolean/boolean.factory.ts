import { ColDef } from "@ag-grid-community/core";
import { TranslateService } from "@ngx-translate/core";

import { ITypeControllerParams } from "../../entity";
import { ColDefBuilder } from "../base/base-abstract-factory/builders/col-def-builder";
import { IColDefAbstractFactory } from "../base/base-abstract-factory/models";
import { MenuTab } from "../base/base-abstract-factory/builder-models";
import { BooleanImageRendererComponent } from "../../../../../cellRenderer/booleanImage/boolean-renderer.component";
import { CellClassOptions, VisibilityOptions, WidthOptions } from "../base/base-abstract-factory/builder-options";
import {
	BooleanFilteringOptions,
	BooleanGroupingOptions,
	BooleanSortingOptions,
	BooleanTooltipOptions,
	BooleanValueFormatterOptions,
	BooleanValueGetterOptions
} from "./builder";

export class BooleanFactorySG implements IColDefAbstractFactory {

	createColDef(controllerParams: ITypeControllerParams): ColDef {
		return new ColDefBuilder()
			.setColId(controllerParams.meta.columnName)
			.setWidth(new WidthOptions())
			.setTooltipOptionsByFunction(new BooleanTooltipOptions(controllerParams.injector.get(TranslateService)))
			.setMenuTabs([MenuTab.GeneralMenuTab, MenuTab.FilterMenuTab, MenuTab.ColumnsMenuTab])
			.setCellClass(new CellClassOptions(controllerParams.meta))
			.setHeaderName(controllerParams.meta.headerName)
			.setVisibility(new VisibilityOptions(controllerParams.meta))
			.setCellRenderer(BooleanImageRendererComponent)
			.setSorting(new BooleanSortingOptions(controllerParams.meta, controllerParams.grid))
			.setValueGetterByFunction(new BooleanValueGetterOptions(controllerParams.meta.columnName))
			.ignoreValueSetter()
			.setValueFormatter(new BooleanValueFormatterOptions(controllerParams.injector.get(TranslateService)))
			.setGrouping(new BooleanGroupingOptions())
			.setCustomFiltering(new BooleanFilteringOptions(controllerParams.meta, controllerParams.grid))
			.build();
	}
}
