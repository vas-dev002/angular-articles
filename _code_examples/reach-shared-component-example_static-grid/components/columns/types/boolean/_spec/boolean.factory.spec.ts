import { ColDef } from "@ag-grid-community/core";
import { Injector } from "@angular/core";
import { TranslateModule, TranslateService } from "@ngx-translate/core";

import { inject, TestBed } from "@angular/core/testing";

import { IColumnMetaSG } from "../../../../main/entity";
import { StaticGridComponent } from "../../../../main/comp";
import { COMMON_META } from "../../base/_spec/common.set";
import { BooleanFactorySG } from "../boolean.factory";

declare const expect: jest.Expect;

describe("Static Grid Column Type Boolean: ", () => {
	const meta: IColumnMetaSG = COMMON_META;
	const customOptions = {};
	const rowData = [
		{ [meta.columnName]: "False" },
		{ [meta.columnName]: "" },
		{ [meta.columnName]: "True" }
	];
	const grid = { rowData } as StaticGridComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [TranslateModule.forRoot()],
			providers: [TranslateService]
		});
	});

	describe(`Controller: BooleanFactorySG `, () => {
		let factory: BooleanFactorySG;
		let columnDefinition: ColDef;
		let injector: Injector;

		beforeEach(inject([Injector], (inj: Injector) => {
			injector = inj;
			factory = new BooleanFactorySG();
		}));

		describe("[Method]: createColDef ", () => {
			it("should return default column definition", () => {
				columnDefinition = factory.createColDef({ meta, grid, customOptions, injector });
				expect(columnDefinition).toMatchSnapshot();
			});
		});
	});
});
