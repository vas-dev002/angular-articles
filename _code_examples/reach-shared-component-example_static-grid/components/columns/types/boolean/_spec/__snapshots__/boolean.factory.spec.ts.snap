// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Grid Column Type Boolean:  Controller: BooleanFactorySG  [Method]: createColDef  should return default column definition 1`] = `
Object {
  "cellClass": "testCss",
  "cellClassRules": Object {
    "sg-cell-disabled": [Function],
  },
  "cellRendererFramework": [Function],
  "colId": "testColumn",
  "comparator": [Function],
  "enableRowGroup": true,
  "field": "testColumn",
  "filterFramework": [Function],
  "filterParams": Object {
    "meta": Object {
      "cellDefaultText": "Unassigned",
      "filterOptions": Array [],
    },
  },
  "floatingFilterComponentFramework": [Function],
  "floatingFilterComponentParams": Object {
    "meta": Object {
      "cellDefaultText": "Unassigned",
      "filterOptions": Array [],
    },
    "suppressFilterButton": true,
  },
  "headerName": "Test Header",
  "hide": true,
  "keyCreator": [Function],
  "lockVisible": false,
  "maxWidth": 500,
  "menuTabs": Array [
    "generalMenuTab",
    "filterMenuTab",
    "columnsMenuTab",
  ],
  "minWidth": 50,
  "sortingOrder": Array [
    "desc",
    "asc",
    null,
  ],
  "tooltipComponentFramework": [Function],
  "tooltipValueGetter": [Function],
  "valueFormatter": [Function],
  "valueGetter": [Function],
  "width": undefined,
}
`;
